### DISCUSSION TITLE
How to set environment variables for debuggable application

### DISCUSSION THREAD

**Discussion by 39george** [mrcjkb/rustaceanvim#443] [Q&A] [answered]
I'm a big fan of using Neovim with the nvim-dap plugin. Debugging works great! In web development, it's common practice to use a .env file to store environment variables.

My question is: can I pass these variables from the .env file into my application during debugging with nvim-dap and rustaceanvim?
I tried setting up launch.json like this:
```json
{
  "version": "0.2.0",
  "configurations": [
    {
      "name": "Debug",
      "type": "lldb",
      "request": "launch",
      "env": {
        "APP_CONFIG_FILE": "config/local.yaml",
        "POSTGRES_USER": "postgres",
        "POSTGRES_PASSWORD_FILE": "secrets/postgres_password.txt",
        "REDIS_PASSWORD_FILE": "secrets/redis_password.txt",
        "EMAIL_AUTHORIZATION_TOKEN_FILE": "secrets/email_client_authorization_token.txt",
        "OBJECT_STORAGE_KEY_ID_FILE": "secrets/object_storage_key_id.txt",
        "OBJECT_STORAGE_ACCESS_KEY_FILE": "secrets/object_storage_access_key.txt",
        "ENVIRONMENT": "development",
      }
    }
  ]
}
```
However, my backend application doesn't seem to be able to access these variables.

**Comment by mrcjkb** [comment, marked as answer]
Hey :wave: 

rustaceanvim only creates nvim-dap configurations from rust-analyzer responses.
You're probably better of asking this in the [nvim-dap discusstions](https://github.com/mfussenegger/nvim-dap/discussions/categories/q-a).

You might also find something in [`:h dap-adapter`](https://github.com/mfussenegger/nvim-dap/blob/6f79b822997f2e8a789c6034e147d42bc6706770/doc/dap.txt#L45).
